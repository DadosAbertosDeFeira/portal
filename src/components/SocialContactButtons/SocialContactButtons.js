import React from 'react';
import { useRouter } from 'next/router';

const SocialContactButtons = () => {
  const router = useRouter();

  return (
    <div className="grid grid-flow-col grid-cols-2 grid-rows-2 auto-cols-auto mt-md gap-md self-center tablet:self-start">
      <button
        data-testid="twitter-button"
        className="bg-blue-100 h-xl px-sm  rounded flex items-center w-4xl justify-center space-x-sm pa-md"
        type="button"
        onClick={() => router.push(process.env.NEXT_PUBLIC_TWITTER_URL)}
      >
        <img src="/assets/icons/twitter.svg" alt="Twitter" />
        <span className="text-primary-dark text-sm">Twitter</span>
      </button>

      <button
        data-testid="facebook-button"
        className="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
        type="button"
        onClick={() => router.push(process.env.NEXT_PUBLIC_FACEBOOK_URL)}
      >
        <img src="/assets/icons/facebook.svg" alt="Facebook" />
        <span className="text-primary-dark text-sm">Facebook</span>
      </button>

      <button
        data-testid="instagram-button"
        className="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
        type="button"
        onClick={() => router.push(process.env.NEXT_PUBLIC_INSTAGRAM_URL)}
      >
        <img src="/assets/icons/instagram.svg" alt="Instagram" />
        <span className="text-primary-dark text-sm">Instagram</span>
      </button>

      <button
        data-testid="github-button"
        className="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
        type="button"
        onClick={() => router.push(process.env.NEXT_PUBLIC_GITHUB_URL)}
      >
        <img src="/assets/icons/github.svg" alt="Github" />
        <span className="text-primary-dark text-sm">Github</span>
      </button>
    </div>
  );
};

export default SocialContactButtons;
