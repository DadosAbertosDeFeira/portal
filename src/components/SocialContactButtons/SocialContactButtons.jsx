import React from "react";
import Image from "next/image";

import Twitter from "../../../public/assets/icons/twitter.svg";
import Facebook from "../../../public/assets/icons/facebook.svg";
import Instagram from "../../../public/assets/icons/instagram.svg";
import Github from "../../../public/assets/icons/github.svg";

const SocialContactButtons = () => {
  const handleClick = (event) => {
    const { url } = event.currentTarget.dataset;
    window.open(url, "_blank").focus();
  };

  return (
    <div className="grid grid-flow-col grid-cols-2 grid-rows-2 auto-cols-auto mt-md gap-md self-center tablet:self-start">
      <button
        data-testid="twitter-button"
        className="bg-blue-100 h-xl px-sm  rounded flex items-center w-4xl justify-center space-x-sm pa-md"
        type="button"
        data-url={process.env.NEXT_PUBLIC_TWITTER_URL}
        onClick={handleClick}
      >
        <Image src={Twitter} alt="Twitter" />
        <span className="text-primary-dark text-sm">Twitter</span>
      </button>

      <button
        data-testid="facebook-button"
        className="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
        type="button"
        data-url={process.env.NEXT_PUBLIC_FACEBOOK_URL}
        onClick={handleClick}
      >
        <Image src={Facebook} alt="Facebook" />
        <span className="text-primary-dark text-sm">Facebook</span>
      </button>

      <button
        data-testid="instagram-button"
        className="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
        type="button"
        data-url={process.env.NEXT_PUBLIC_INSTAGRAM_URL}
        onClick={handleClick}
      >
        <Image src={Instagram} alt="Instagram" />
        <span className="text-primary-dark text-sm">Instagram</span>
      </button>

      <button
        data-testid="github-button"
        className="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
        type="button"
        data-url={process.env.NEXT_PUBLIC_GITHUB_URL}
        onClick={handleClick}
      >
        <Image src={Github} alt="Github" />
        <span className="text-primary-dark text-sm">Github</span>
      </button>
    </div>
  );
};

export default SocialContactButtons;
