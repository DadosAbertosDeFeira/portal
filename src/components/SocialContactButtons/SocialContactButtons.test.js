import React from 'react';
import { useRouter } from 'next/router';
import { render, fireEvent } from '@testing-library/react';

import SocialContactButtons from './SocialContactButtons';

jest.mock('next/router');

describe('<SocialContactButtons />', () => {
  it('renders correctly all the buttons', () => {
    const { getByText, asFragment, getByAltText } = render(
      <SocialContactButtons />
    );

    expect(getByText('Twitter')).toBeInTheDocument();
    expect(getByText('Facebook')).toBeInTheDocument();
    expect(getByText('Instagram')).toBeInTheDocument();
    expect(getByText('Github')).toBeInTheDocument();

    expect(getByAltText('Twitter')).toBeInTheDocument();
    expect(getByAltText('Facebook')).toBeInTheDocument();
    expect(getByAltText('Instagram')).toBeInTheDocument();
    expect(getByAltText('Github')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  describe('behavior', () => {
    let routerMock;

    beforeEach(() => {
      jest.resetModules();
      jest.resetAllMocks();
      routerMock = { push: jest.fn() };
      useRouter.mockReturnValue(routerMock);
    });

    it('redirects to Twitter when user clicks on the button', () => {
      const { getByTestId } = render(<SocialContactButtons />);
      fireEvent.click(getByTestId('twitter-button'));

      expect(routerMock.push).toHaveBeenCalledWith(
        'https://twitter.com/DadosDeFeira'
      );
    });

    it('redirects to Facebook when user clicks on the button', () => {
      const { getByTestId } = render(<SocialContactButtons />);
      fireEvent.click(getByTestId('facebook-button'));

      expect(routerMock.push).toHaveBeenCalledWith(
        'https://www.facebook.com/dadosabertosdefeira'
      );
    });

    it('redirects to Instagram when user clicks on the button', () => {
      const { getByTestId } = render(<SocialContactButtons />);
      fireEvent.click(getByTestId('instagram-button'));

      expect(routerMock.push).toHaveBeenCalledWith(
        'https://www.instagram.com/dadosabertosdefeira/'
      );
    });

    it('redirects to Github when user clicks on the button', () => {
      const { getByTestId } = render(<SocialContactButtons />);
      fireEvent.click(getByTestId('github-button'));

      expect(routerMock.push).toHaveBeenCalledWith(
        'https://github.com/DadosAbertosDeFeira'
      );
    });
  });
});
