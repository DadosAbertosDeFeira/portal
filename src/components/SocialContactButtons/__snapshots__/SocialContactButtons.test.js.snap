// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialContactButtons /> renders correctly all the buttons 1`] = `
<DocumentFragment>
  <div
    class="grid grid-flow-col grid-cols-2 grid-rows-2 auto-cols-auto mt-md gap-md self-center tablet:self-start"
  >
    <button
      class="bg-blue-100 h-xl px-sm  rounded flex items-center w-4xl justify-center space-x-sm pa-md"
      data-testid="twitter-button"
      type="button"
    >
      <img
        alt="Twitter"
        src="/assets/icons/twitter.svg"
      />
      <span
        class="text-primary-dark text-sm"
      >
        Twitter
      </span>
    </button>
    <button
      class="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
      data-testid="facebook-button"
      type="button"
    >
      <img
        alt="Facebook"
        src="/assets/icons/facebook.svg"
      />
      <span
        class="text-primary-dark text-sm"
      >
        Facebook
      </span>
    </button>
    <button
      class="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
      data-testid="instagram-button"
      type="button"
    >
      <img
        alt="Instagram"
        src="/assets/icons/instagram.svg"
      />
      <span
        class="text-primary-dark text-sm"
      >
        Instagram
      </span>
    </button>
    <button
      class="bg-blue-100 h-xl px-sm rounded flex items-center w-4xl justify-center space-x-sm pa-md "
      data-testid="github-button"
      type="button"
    >
      <img
        alt="Github"
        src="/assets/icons/github.svg"
      />
      <span
        class="text-primary-dark text-sm"
      >
        Github
      </span>
    </button>
  </div>
</DocumentFragment>
`;
